app/javascript/components/rules/RelatedRulesModal.vue

```diff
@@ -422,12 +422,13 @@ export default {
     },
     lookupSearchWordInRules: function (rules) {
       const words = this.keywordList.map((w) => w.toLowerCase());
-      const checkWord = (text) => words.some((w) => text.includes(w));
+      const checkWord = (text) => words.some((w) => text ? text.includes(w) : '');
```

Comment:
refactor: `const checkWord = (text) => text && words.some((w) => text.includes(w));`

this ensures that if text is not provided (i.e., it's falsy like null, undefined, or an empty string), the function returns false without iterating through the words array.
will push the change